celi_framework/utils/llms.py

```diff
@@ -223,9 +229,9 @@ async def quick_ask_async(
                 )
 
             if json_output:
+                if response_format:
+                    app_logger.warning(f"A value has been provided for response_format while json_output is enabled. JSON mode will be use and response_format will be ignored.")
```

Comment:
Why not honor the response_format here?  It's not a backwards compatibility problem, since any existing code won't have response_format.  

I'm worried this approach as written will lead to people not seeing a warning buried in the output and will waste time tracking down why their response_format isn't working.  Seems like it would be better to honor that.

Reply:
So instead of defaulting to JSON mode default to Structured Outputs? It's an either/or situation, so we have to pick one or the other. 

JSON mode will enforce a valid JSON structure, whereas Structured Outputs will strictly force a specific JSON structure the user has provided. In the former case the LLM can decide the structure. In the latter case you have to explicitly provide the structure.

Reply:
I'm proposing that if they pass both we enforce the structured outputs (which implicitly provides valid JSON).  I think this is the most intuitive behavior.
Ignoring the response_fortmat is json_output is set seems likely to have unintuitive behavior for the user

Reply:
Agreed. I will make that change.

I would still include a warning noting the conflicting arguments. I believe JSON mode could still be useful if you don't know exactly what structure you want and want the model to come up with it for you. It seems very unlikely someone will provide an argument for response_format if they want the model to come up with the structure, but we should probably still flag it.